recipes_source/bundled_inputs.rst

```diff
@@ -0,0 +1,193 @@
+-------------------
+1. **Prepare model**: Convert your model to TorchScript through either tracing or scripting
+
+.. code:: python
+
+    import torch
+    import torch.jit
+    import torch.utils
+    import torch.utils.bundled_inputs
+
+    class Net(nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.lin = nn.Linear(10, 1)
+
+        def forward(self, x):
+            return self.lin(x)
+
+    model = Net()
+    scripted_module = torch.jit.script(model)
+
+2. **Create example input and attach to model**
+
+.. code:: python
+
+    # For each method create a list of inputs and each input is a tuple of arguments
+    input = [(torch.zeros(1,10),)]
+
+    # Create model with bundled inputs, if type(input) is list then the input is bundled to 'forward'
+    bundled_model = bundle_inputs(scripted_module, input)
+
+
+3. **Retrieve inputs and run model on them**
+
+.. code:: python
+    inputs = bundled_model.get_all_bundled_inputs()
+
+    print(bundled_model(*inputs[0]))
+
+
+Uncommon case bundling and retrieving inputs for functions beyond 'forward'
+-------------------
+1. **Prepare model**: Convert your model to TorchScript through either tracing or scripting
+
+.. code:: python
+
+    import torch
+    import torch.jit
+    import torch.utils
+    import torch.utils.bundled_inputs
+    from typing import Dict
+
+    class Net(nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.lin = nn.Linear(10, 1)
+
+        def forward(self, x):
+            return self.lin(x)
+
+        @torch.jit.export
+        def foo(self, x: Dict[String, int]):
+            return x['a'] + x['b']
+
+
+    model = Net()
+    scripted_module = torch.jit.script(model)
+
+2. **Create example input and attach to model**
+
+.. code:: python
+
+    # For each method create a list of inputs and each input is a tuple of arguments
+    example_dict = {'a' : 1, 'b' : 2}
+    input = {
+        scripted_module.forward : [(torch.zeros(1,10),)],
+        scripted_module.foo : [(example_dict,)]
+    }
+
+    # Create model with bundled inputs, if type(input) is Dict then each callable key is mapped to its corresponding bundled input
+    bundled_model = bundle_inputs(scripted_module, input)
+
+
+3. **Retrieve inputs and run model on them**
+
+.. code:: python
+
+    all_info = bundled_model.get_bundled_inputs_functions_and_info()
+
+    # The return type for get_bundled_inputs_functions_and_info is complex, but essentially we are retrieving the name
+    # of a function we can use to get the bundled input for our models method
+    for func_name in all_info.keys():
+        input_func_name = all_info[func_name]['get_inputs_function_name'][0]
+        func_to_run = getattr(bundled_model, input_func_name)
+        # retrieve input
+        input = func_to_run()
+        model_function = getattr(bundled_model, func_name)
+        print(model_function(*input[0]))
```

Comment:
Maybe document that `input[0]` is the first bundled input or update this code to loop over all inputs?